### ISSUE TITLE
Ensure that //drake/bindings do not cause cpplint issues after clang-format

### ISSUE THREAD

**Issue by EricCousineau-TRI** [RobotLocomotion/drake#5968]
At present, `//drake/bindings/pybind11` do not appear to conform to the formatting produced by `clang-format` (all `*.cc` files are modified); however, when `clang-format` changes them, they are then invalid by `cpplint`.

I believe the short-term solution would be to `clang-format` the files, manually fix the errors, and ensure that further applying `clang-format` does not change them.

Will do after #5890.

\cc @soonho-tri 

```
$ git ref
126a584
$ git status -b
$ find . -name '*.cc' -o -name '*.h'
./pydrake_symbolic.cc
./pydrake_symbolic_types.h
./pydrake_rbtree.cc
./pydrake_autodiffutils.cc
./pydrake_ik.cc
./pydrake_mathematicalprogram.cc
./pydrake_mosek.cc
./pydrake_gurobi.cc
./pydrake_parsers.cc
./pydrake_autodiff_types.h
$ find . -name '*.cc' -o -name '*.h' | xargs clang-format -i
$ git status -b
 M pydrake_autodiffutils.cc
 M pydrake_gurobi.cc
 M pydrake_ik.cc
 M pydrake_mathematicalprogram.cc
 M pydrake_mosek.cc
 M pydrake_parsers.cc
 M pydrake_rbtree.cc
 M pydrake_symbolic.cc
 M pydrake_symbolic_types.h
$ drake-cpplint 
...
 TOTAL 1330 files checked, found 11 warnings
.../drake/bindings/pybind11/pydrake_rbtree.cc:75: warning: Missing space before {  [whitespace/braces] [5]
.../drake/bindings/pybind11/pydrake_ik.cc:20: warning: Extra space before ( in function call  [whitespace/parens] [4]
.../drake/bindings/pybind11/pydrake_ik.cc:78: warning: Extra space before ( in function call  [whitespace/parens] [4]
.../drake/bindings/pybind11/pydrake_mathematicalprogram.cc:74: warning: Extra space before ( in function call  [whitespace/parens] [4]
.../drake/bindings/pybind11/pydrake_mathematicalprogram.cc:79: warning: Extra space before ( in function call  [whitespace/parens] [4]
.../drake/bindings/pybind11/pydrake_mathematicalprogram.cc:84: warning: Extra space before ( in function call  [whitespace/parens] [4]
.../drake/bindings/pybind11/pydrake_mathematicalprogram.cc:89: warning: Extra space before ( in function call  [whitespace/parens] [4]
.../drake/bindings/pybind11/pydrake_mathematicalprogram.cc:140: warning: Extra space before ( in function call  [whitespace/parens] [4]
.../drake/bindings/pybind11/pydrake_mathematicalprogram.cc:144: warning: Extra space before ( in function call  [whitespace/parens] [4]
.../drake/bindings/pybind11/pydrake_mathematicalprogram.cc:147: warning: Extra space before ( in function call  [whitespace/parens] [4]
.../drake/bindings/pybind11/pydrake_mathematicalprogram.cc:150: warning: Extra space before ( in function call  [whitespace/parens] [4]
```

**Comment by jwnimmer-tri** [comment]
No longer relevant, since we're mandating clang-format idempotency now, so it will always either work or fail without surprising people with regressions on master.
